### ISSUE TITLE
Decimal class shenanigans causing failed validation

### ISSUE THREAD

**Issue by TheMule71** [sissaschool/xmlschema#213]
```
import xmlschema

schema = xmlschema.XMLSchema("""
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
<xs:element name="amount" type="xs:decimal"/>
</xs:schema>
""")

xml1 = """<?xml version="1.0" encoding="UTF-8"?><amount>0.000000</amount>"""
print(schema.to_dict(xml1))

xml2 = """<?xml version="1.0" encoding="UTF-8"?><amount>0.0000000</amount>"""
print(schema.to_dict(xml2))
```
raises:
```
xmlschema.validators.exceptions.XMLSchemaValidationError: failed validating Decimal('0E-7') with <function decimal_validator at 0x7fd1e46cb790>:

Reason: value Decimal('0E-7') is not a valid xs:decimal

Instance:

  <amount>0.0000000</amount>

Path: /amount
```

decimal.Decimal() is representing 0.000000 as 0.000000 but 0.0000000 as 0E-7.

Is there a way to prevent this? Like overriding the class used internally to represent the value. I've tried with decimal_type=str, passing my converter class, playing with global_maps= and validator.maps to no avail.

**Comment by brunato** [comment]
I will change xs:decimal built-in type using `elementpath.datatypes.DecimalProxy` for creating python Decimal objects. Furthemore i will use _DecimalProxy_ in decimal_validator() function (but admitting also float values for extending re-encoding if `decimal_type=float` is used). 
The fix should be available in next release.

Thank you

**Comment by TheMule71** [comment]
Great!
Let me know if you need me to test it.

FYI, we're about to use xmlschema here: https://github.com/OCA/l10n-italy/pull/1958. Looking good so far.

**Comment by brunato** [comment]
> Let me know if you need me to test it.
> 
Sure, i put a test fot this but i will notify you for a confirmation check before closing the issue.

> FYI, we're about to use xmlschema here: [OCA/l10n-italy#1958](https://github.com/OCA/l10n-italy/pull/1958). Looking good so far.

Wonderful!
I already use this for validate pagoPA messages in a Django site, I will add also fattura-pa schemas in my extra tests repository ...

**Comment by MarcoColombo71** [comment]
This is the official xsd from fatturapa: https://www.fatturapa.gov.it/export/documenti/fatturapa/v1.2.1/Schema_del_file_xml_FatturaPA_versione_1.2.1a.xsd
we had to modify it as xmlschema was complaining about entities in an external file. See this commit: https://github.com/OCA/l10n-italy/pull/1958/commits/167340d774e8c0390b15d8528f2683c95038dfd7

We perform a number of tests, if interested you can find them here: https://github.com/TheMule71/l10n-italy/tree/12.0-dev-xmlschema/l10n_it_fatturapa_in/tests
As of now, they all pass, with xmlschema.
https://github.com/TheMule71/l10n-italy/runs/1432834568

**Comment by TheMule71** [comment]
This is the official xsd from fatturapa: https://www.fatturapa.gov.it/export/documenti/fatturapa/v1.2.1/Schema_del_file_xml_FatturaPA_versione_1.2.1a.xsd
we had to modify it as xmlschema was complaining about entities in an external file. See this commit: OCA/l10n-italy@167340d

We perform a number of tests, if interested you can find them here: https://github.com/TheMule71/l10n-italy/tree/12.0-dev-xmlschema/l10n_it_fatturapa_in/tests
As of now, they all pass, with xmlschema.
https://github.com/TheMule71/l10n-italy/runs/1432834568

**Comment by brunato** [comment]
Hi, the fix is available with release v1.4.0.
